avro2py/avro_types.py

```diff
@@ -175,6 +176,7 @@ def parse_into_types(schema: Union[Dict[str, Any], str, List], parent_namespace:
     if type_name == 'record':
         return Record(
             original_schema=schema,
+            resolved_schema=schema,
```

Comment:
This isn't going to work if these are the same schema, because when we mutate `resolved_schema` (which we do below), it's the same object that's going to be mutated. We either need to do a deepcopy on one of these, _not_ use mutation in `resolve_schema_reference` below, or get rid of `original_schema` entirely, if it's no longer being used.

Reply:
I have used `deepcopy` for now.
My intention was to keep `original_schema` around if required for whatever reason later.
`original_schema` is not being used currently. We can delete it as well.